src/Randomizer.App/SpriteService.cs

```diff
@@ -0,0 +1,184 @@
+
+            var extraSpriteDirectory = Environment.ExpandEnvironmentVariables("%LocalAppData%\\SMZ3CasRandomizer\\Sprites");
+
+            if (Directory.Exists(extraSpriteDirectory))
+            {
+                sprites.AddRange(Directory.EnumerateFiles(extraSpriteDirectory, "*.rdc", SearchOption.AllDirectories)
+                    .Select(LoadRdcSprite));
+                sprites.AddRange(Directory.EnumerateFiles(extraSpriteDirectory, "*.ips", SearchOption.AllDirectories)
+                    .Select(LoadIpsSprite));
+            }
+
+            Sprites = sprites;
+
+            _logger.LogInformation("{LinkSprites} Link Sprites | {SamusSprites} Samus Sprites | {ShipCount} Ship Sprites", LinkSprites.Count(), SamusSprites.Count(), ShipSprites.Count());
+        });
+    }
+
+    /// <summary>
+    /// Retrieves the random sprite image for the given sprite type
+    /// </summary>
+    /// <param name="type"></param>
+    /// <returns></returns>
+    public string GetRandomPreviewImage(SpriteType type)
+    {
+        var spriteFolder = type == SpriteType.Ship ? "Ships" : type.ToString();
+        return Path.Combine(SpritePath, spriteFolder, "random.png");
+    }
+
+    /// <summary>
+    /// Loads a Link or Samus rdc sprite
+    /// </summary>
+    /// <param name="path">The path to the rdc file</param>
+    /// <returns>The Sprite object</returns>
+    private Sprite LoadRdcSprite(string path)
+    {
+        using var stream = File.OpenRead(path);
+        var rdc = Rdc.Parse(stream);
+
+        var spriteType = SpriteType.Unknown;
+        if (rdc.Contains<LinkSprite>())
+            spriteType = SpriteType.Link;
+        else if (rdc.Contains<SamusSprite>())
+            spriteType = SpriteType.Samus;
+
+        var name = Path.GetFileName(path);
+        var author = rdc.Author;
+        if (rdc.TryParse<MetaDataBlock>(stream, out var block))
+        {
+            var title = block?.Content?.Value<string>("title");
+            if (!string.IsNullOrEmpty(title))
+                name = title;
+
+            var author2 = block?.Content?.Value<string>("author");
+            if (string.IsNullOrEmpty(author) && !string.IsNullOrEmpty(author2))
+                author = author2;
+        }
+
+        var file = new FileInfo(path);
+        var previewPath = file.FullName.Replace(file.Extension, ".png");
+        if (!File.Exists(previewPath))
+        {
+            previewPath = GetRandomPreviewImage(spriteType);
+        }
+
+        return new Sprite(name, author, path, spriteType, previewPath);
+    }
+
+    /// <summary>
+    /// Loads a ship ips sprite
+    /// </summary>
+    /// <param name="path">The path to the ips file</param>
+    /// <returns>The sprite object</returns>
+    private Sprite LoadIpsSprite(string path)
+    {
+        var file = new FileInfo(path);
+        var parts = Path.GetFileNameWithoutExtension(path).Split(" by ");
+        var name = parts[0];
+        var author = parts[1];
```

Comment:
This'll throw if someone drops in an incorrectly named file. Did you handle that elsewhere?